### ISSUE TITLE
%s process (%s) instead of a process name

### ISSUE THREAD

**Issue by akansc** [rspamd/rspamd#4634]
With the latest version - the rspamd started to show "%s process (%s)" instead of a process name on rhel systems.
Kind of messed up with our monitoring.

### Steps to Reproduce

Update to the latest rspamd version on a rhel system.

Examples:
1.
```
[root@server1 ~]# cat /etc/redhat-release
CloudLinux release 7.9 (Boris Yegorov)


[root@server1 ~]# rspamd -v
Rspamd daemon version 3.7.1

CPU architecture x86_64; features: sse2, sse3
Hyperscan enabled: TRUE
Jemalloc enabled: TRUE
LuaJIT enabled: TRUE (LuaJIT version: LuaJIT 2.1.1695653777)
ASAN enabled: FALSE
BLAS enabled: TRUE
Fasttext enabled: FALSE


[root@server1 ~]# /usr/bin/ps -eo 's uid pid ppid vsz rss pcpu etime comm args' | grep rspamd
S   988  781643       1 361064 50112  0.0       08:43 main process    /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   988  781856  781643 361064 42872  0.0       08:43 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   988  781858  781643 361972 43864  0.0       08:43 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   988  781859  781643 361192 43312  0.0       08:43 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   988  781860  781643 361192 43292  0.0       08:43 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   988  781862  781643 361192 43292  0.0       08:43 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   988  781864  781643 361192 43292  0.0       08:43 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
```

2.
```
[root@server2 rspamd]# cat /etc/redhat-release
CloudLinux release 8.8 (Anatoly Filipchenko)


[root@server2 ~]# rspamd -v
Rspamd daemon version 3.7.1

CPU architecture x86_64; features: avx2, avx, sse2, sse3, ssse3, sse4.1, sse4.2, rdrand
Hyperscan enabled: TRUE
Jemalloc enabled: TRUE
LuaJIT enabled: TRUE (LuaJIT version: LuaJIT 2.1.1695653777)
ASAN enabled: FALSE
BLAS enabled: TRUE
Fasttext enabled: FALSE


[root@server2 ~]# /usr/bin/ps -eo 's uid pid ppid vsz rss pcpu etime comm args' | grep rspamd
S   976 2660557       1 677592 296976 0.6       39:25 %s              /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   976 2661559 2660557 677592 277456 0.0       39:10 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   976 2661560 2660557 678372 124136 0.0       39:10 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   976 2661561 2660557 677592 133176 0.2       39:10 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   976 2661563 2660557 677720 130716 0.1       39:10 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   976 2661564 2660557 677912 130832 0.5       39:10 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   976 2661565 2660557 677912 133984 0.8       39:10 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   976 2661566 2660557 677592 274228 0.0       39:10 %s process      /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
```

3.
```
[root@server ~]# cat /etc/redhat-release
CentOS Linux release 7.9.2009 (Core)


[root@server ~]# rspamd -v
Rspamd daemon version 3.7.1

CPU architecture x86_64; features: avx2, avx, sse2, sse3, ssse3, sse4.1, sse4.2, rdrand
Hyperscan enabled: TRUE
Jemalloc enabled: TRUE
LuaJIT enabled: TRUE (LuaJIT version: LuaJIT 2.1.1695653777)
ASAN enabled: FALSE
BLAS enabled: TRUE
Fasttext enabled: FALSE


[root@server ~]# /usr/bin/ps -eo 's uid pid ppid vsz rss pcpu etime comm args' | grep rspamd
S   985   679     1 670916 96160 73.0       00:24 main process    /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   985   803   679 669840 83936  0.0       00:06 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   985   804   679 671824 86128  0.1       00:06 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   985   805   679 671044 85844  0.1       00:06 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   985   806   679 671044 85836  0.1       00:06 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   985   807   679 671044 85836  0.1       00:06 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   985   808   679 671044 85836  0.1       00:06 %s process (%s) /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
S   985   809   679 669712 88484 14.1       00:06 %s process      /usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
```

### Expected output with the process name as on older versions of rspamd

```
[root@server ~]# /usr/bin/ps -eo 's uid pid ppid vsz rss pcpu etime comm args' | grep rspamd
S   985 15281     1 671532 87500  0.0    13:35:33 rspamd          rspamd: main process; 0.1 msg/sec, 0.0 msg/sec spam, 0.1 msg/sec ham; 2.22s avg processing time
S   985 15387 15281 670328 83940  0.0    13:35:16 rspamd          rspamd: rspamd_proxy process (/var/run/rspamd/rspamd_proxy.sock mode=0600 owner=_rspamd)
S   985 15388 15281 675704 90616  0.0    13:35:16 rspamd          rspamd: controller process (/var/run/rspamd/rspamd_controller.sock mode=0600 owner=_rspamd)
S   985 15389 15281 676184 95404  0.0    13:35:16 rspamd          rspamd: normal process (localhost:11333)
S   985 15390 15281 676396 96284  0.0    13:35:16 rspamd          rspamd: normal process (localhost:11333)
S   985 15391 15281 676440 95976  0.0    13:35:16 rspamd          rspamd: normal process (localhost:11333)
S   985 15392 15281 677016 96772  0.0    13:35:16 rspamd          rspamd: normal process (localhost:11333)
S   985 15393 15281 670328 84668  0.0    13:35:16 rspamd          rspamd: hs_helper process
```

**Comment by vstakhov** [comment]
It seems that `setproctitle` is used from some other library.

**Comment by vstakhov** [comment]
Is it a regression from 3.6?

**Comment by akansc** [comment]
Yes, the previous version was:
```
[root@server1 ~]#  /usr/bin/ps -eo 's uid pid ppid vsz rss pcpu etime comm args' | grep rspamd
S   988  795984       1 353488 49688  0.0    14:51:29 rspamd          rspamd: main process
S   988  795985  795984 353616 42420  0.0    14:51:28 rspamd          rspamd: rspamd_proxy process (/var/run/rspamd/rspamd_proxy.sock mode=0600 owner=_rspamd)
S   988  795986  795984 358236 50676  0.0    14:51:28 rspamd          rspamd: controller process (/var/run/rspamd/rspamd_controller.sock mode=0600 owner=_rspamd)
S   988  795987  795984 354384 44644  0.0    14:51:28 rspamd          rspamd: normal process (localhost:11333)
S   988  795988  795984 354384 44644  0.0    14:51:28 rspamd          rspamd: normal process (localhost:11333)
S   988  795989  795984 354384 44648  0.0    14:51:28 rspamd          rspamd: normal process (localhost:11333)
S   988  795990  795984 354384 44644  0.0    14:51:28 rspamd          rspamd: normal process (localhost:11333)


[root@server1 ~]# rspamd -v
Rspamd daemon version 3.6

CPU architecture x86_64; features: sse2, sse3
Hyperscan enabled: TRUE
Jemalloc enabled: TRUE
LuaJIT enabled: TRUE (LuaJIT version: LuaJIT 2.1.0-beta3)
ASAN enabled: FALSE
BLAS enabled: TRUE
Fasttext enabled: FALSE


[root@server1 ~]# rpm -qa | grep rspam
rspamd-3.6-2.el7.x86_64
```

**Comment by akansc** [comment]
This issue seem to have broke DirectAdmin Rspamd process detection as well on rhel based systems. Spam related functionality on user level stopped working unless we downgrade the Rspamd to the previous 3.6.2 version.

**Comment by sunitsonavane** [comment]
@akansc mentioned. We experienced the same and downgrading to 3.6.2 solved it for us.

**Comment by vstakhov** [comment]
Do you use my packages?

**Comment by akansc** [comment]
Yes. If you enable Rspamd on a directadmin server, it uses your repo by default:

```
[root@server ~]# cat /etc/yum.repos.d/rspamd.repo
[rspamd]
name=Rspamd stable repository
baseurl=http://rspamd.com/rpm-stable/centos-8/$basearch
enabled=1
gpgcheck=1
repo_gpgcheck=1
gpgkey=http://rspamd.com/rpm/gpg.key
```

So it kind of stopped working on all the servers that recently run `yum update`.
The same issue is on rhel7 and rhel8 based systems, I do not have anything near with rhel9 by hand to verify that one.

Interesting enough - Debian packages from your repo doesn't seem to have this issue.

**Comment by cyberhouse6** [comment]
i had same issue on almalinux 8 with directadmin fix was to downgrade to older version

**Comment by fatalbanana** [comment]
> i had same issue on almalinux 8 with directadmin fix was to downgrade to older version

It should be fixed as of 3.7.4; it looks good to me.

**Comment by akansc** [comment]
I can confirm that on my end the 3.7.4 version works as expected:
```
[root@server ~]# rspamd -v
Rspamd daemon version 3.7.4

CPU architecture x86_64; features: sse2, sse3
Hyperscan enabled: TRUE
Jemalloc enabled: TRUE
LuaJIT enabled: TRUE (LuaJIT version: LuaJIT 2.1.1700008891)
ASAN enabled: FALSE
BLAS enabled: TRUE
Fasttext enabled: TRUE


[root@server ~]# /usr/bin/ps -eo 's uid pid ppid vsz rss pcpu etime comm args' | grep rspam[d]
S   988    1101       1 284148  4912  0.0    06:00:38 redis-server    /usr/local/bin/redis-server unixsocket:/var/lib/rspamd/.redis/redis.sock
S   988   30605       1 361248 50236  0.1       03:02 rspamd          rspamd: main process
S   988   30606   30605 361376 42752  0.0       03:02 rspamd          rspamd: rspamd_proxy process (/var/run/rspamd/rspamd_proxy.sock mode=0600 owner=_rspamd)
S   988   30607   30605 362156 43796  0.0       03:02 rspamd          rspamd: controller process (/var/run/rspamd/rspamd_controller.sock mode=0600 owner=_rspamd)
S   988   30608   30605 361376 43336  0.0       03:02 rspamd          rspamd: normal process (localhost:11333)
S   988   30609   30605 361376 43324  0.0       03:02 rspamd          rspamd: normal process (localhost:11333)
S   988   30610   30605 361376 43324  0.0       03:02 rspamd          rspamd: normal process (localhost:11333)
S   988   30611   30605 361376 43324  0.0       03:02 rspamd          rspamd: normal process (localhost:11333)
```
